consensus/core/src/storage/mod.rs

```diff
@@ -30,8 +30,8 @@ pub(crate) trait Store: Send + Sync {
     fn contains_blocks(&self, refs: &[BlockRef]) -> ConsensusResult<Vec<bool>>;
 
     /// Checks whether there is any block at the given slot
-    #[allow(dead_code)]
-    fn contains_block_at_slot(&self, slot: Slot) -> ConsensusResult<bool>;
+    #[cfg(test)]
+    fn contains_block_at_slot(&self, slot: crate::block::Slot) -> ConsensusResult<bool>;
```

Comment:
I feel like `#[expect(dead_code)]` would be more future-proof. This would break if we start actually needing one of them outside of tests.

Reply:
Yes but that's exactly what expect does but better since it's not even compiled in those cases

Reply:
Tbh I actually feel like `allow` is the safe option here

Reply:
The whole point of expect is to notify when something is used, and the test gate works like that but better, so I disagree

Reply:
You are limiting yourself to the current context of these methods being used only for testing purposes. What if we start using them for non testing purposes? Is this breaking change yes or no? I think it is, then I would keep allow.

Reply:
No, it would not be breaking, because it is additive.

Reply:
Oh, I forgot this was a trait method